tdms/src/interpolate.cpp

```diff
@@ -1393,3 +1272,177 @@ void interpolateTimeDomainFieldCentralE_TM(  double ***Exy, double ***Exz, doubl
   *Ey = 0.;
   *Ez = Ezy[k][j][i]+Ezx[k][j][i];
 }
+
+/**
+ * @brief Determines the appropriate interpolation scheme to use
+ * 
+ * @param[in] cells_in_direction The number of Yee cells in the interpolation direction of interest
+ * @param[in] cell_id The current ID (in this dimension) of the Yee cell
+ * @return int Equal to BAND_LIMITED, INTERP1, INTERP2, INTERP3, indicating the appropriate interpolation scheme
+ */
+int determineInterpScheme(int cells_in_direction, int cell_id) {
```

Comment:
```suggestion
InterpolationScheme determineInterpScheme(int cells_in_direction, int cell_id) {
```
where `InterpolationScheme` is an enum? may be slightly more readable

Reply:
or, maybe even better [return a function](https://stackoverflow.com/questions/31387238/c-function-returning-function) (provided the signatures are the same)?

Reply:
Have turned this into an enum. There's scope to turn it into a function pointer but band-limited interpolation takes a different number of arguments to the `interp` functions, and function pointers are one of my least favourite things. Defo something that can be done once I've got band-limited working though